substrate/frame/treasury/src/lib.rs

```diff
@@ -122,6 +122,8 @@ pub type NegativeImbalanceOf<T, I = ()> = <<T as Config<I>>::Currency as Currenc
 >>::NegativeImbalance;
 type AccountIdLookupOf<T> = <<T as frame_system::Config>::Lookup as StaticLookup>::Source;
 type BeneficiaryLookupOf<T, I> = <<T as Config<I>>::BeneficiaryLookup as StaticLookup>::Source;
+pub type BlockNumberFor<T, I = ()> =
```

Comment:
I would rather come up with new name, and not reuse the one which is used for local block number type

Reply:
> I would rather come up with new name, and not reuse the one which is used for local block number type

I am actually doing it the other way around calling the local one as `SystemBlockNumberFor`.

Reply:
yes but this forces you to not import `use frame_system::pallet_prelude::*;`

I think having a different name that doesn't conflict with `frame_system::BlockNumberFor` would be a bit cleaner.

maybe like `ProvidedBlockNumberFor`? I agree the name is not good either.

anyway everything is good to me.

Reply:
> yes but this forces you to not import `use frame_system::pallet_prelude::*;`

Yeah, my hope is to be able to provide a different type in the prelude once enough pallets have been migrated.